src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -353,7 +340,10 @@ private static void WriteTo(TextWriter writer, HtmlEncoder encoder, string value
         {
             if (!string.IsNullOrEmpty(value))
             {
-                encoder.Encode(writer, value);
+                // Perf: Encode write away instead of writing it character-by-character.
+                // character-by-character isn't efficient when using a writer backed by a ViewBuffer.
+                var encoded = encoder.Encode(value);
+                writer.Write(encoded);
```

Comment:
Yeah, that was what I had in mind, but it was just an idea. Not going to tack on more things here. :see_no_evil: